gratipay/models/participant/__init__.py

```diff
@@ -1135,8 +1135,11 @@ def change_username(self, suggested):
 
         lowercased = suggested.lower()
 
-        if lowercased in gratipay.RESTRICTED_USERNAMES:
-            raise UsernameIsRestricted(suggested)
+        # Don't allow any username which is the name of a
+        # file existing on the web_root folder.
+        for ext in ['', '.spt', '.ico', '.txt', '.osdd']:
+            if (lowercased + ext) in gratipay.RESTRICTED_USERNAMES:
+                raise UsernameIsRestricted(suggested)
```

Comment:
What if we took a different approach entirely, and actually ran `/newusername` through the Aspen request processor? That would match the condition more exactly, and would be more future proof.